spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/DefaultEnvironmentPostProcessor.java

```diff
@@ -58,7 +64,17 @@ public void postProcessEnvironment(ConfigurableEnvironment environment, SpringAp
 
 		contributeDefaults(internalDefaults, serverDefaultsResource);
 		contributeDefaults(defaults, serverResource);
-
+		String hostName = null;
+		try {
+			hostName = InetAddress.getLocalHost().getHostAddress();
+		}
+		catch (UnknownHostException e) {
+			// ignore
+		}
+		String dataflowServerPort = environment.getProperty("server.port");
+		if (StringUtils.hasText(hostName) && StringUtils.hasText(dataflowServerPort)) {
+			defaults.put(DataFlowPropertyKeys.PREFIX + "server.uri", String.format("http://%s:%s", hostName, dataflowServerPort));
```

Comment:
for k8s I'm planning on setting env var SPRING_CLOUD_DATAFLOW_SERVER_URI=$SCDF_SERVICE_HOST:$SCDF_SERVICE_PORT